photutils/psf/photometry.py

```diff
@@ -407,3 +355,361 @@ def _model_params2table(self, fit_model, star_group):
+        prepare any 2D model to match this assumption.
+    fitshape : int or length-2 array-like
+        Rectangular shape around the center of a star which will be used
+        to collect the data to do the fitting. Can be an integer to be
+        the same along both axes. E.g., 5 is the same as (5, 5), which
+        means to fit only at the following relative pixel positions:
+        [-2, -1, 0, 1, 2].  Each element of ``fitshape`` must be an odd
+        number.
+    finder : callable or instance of any `~photutils.detection.StarFinderBase` subclasses
+        ``finder`` should be able to identify stars, i.e. compute a
+        rough estimate of the centroids, in a given 2D image.
+        ``finder`` receives as input a 2D image and returns an
+        `~astropy.table.Table` object which contains columns with names:
+        ``id``, ``xcentroid``, ``ycentroid``, and ``flux``. In which
+        ``id`` is an integer-valued column starting from ``1``,
+        ``xcentroid`` and ``ycentroid`` are center position estimates of
+        the sources and ``flux`` contains flux estimates of the sources.
+        See, e.g., `~photutils.detection.DAOStarFinder`.  If ``finder``
+        is ``None``, initial guesses for positions of objects must be
+        provided.
+    fitter : `~astropy.modeling.fitting.Fitter` instance
+        Fitter object used to compute the optimized centroid positions
+        and/or flux of the identified sources. See
+        `~astropy.modeling.fitting` for more details on fitters.
+    aperture_radius : float
+        The radius (in units of pixels) used to compute initial
+        estimates for the fluxes of sources. If ``None``, one FWHM will
+        be used if it can be determined from the ```psf_model``.
+    niters : int or None
+        Number of iterations to perform of the loop FIND, GROUP,
+        SUBTRACT, NSTAR. If None, iterations will proceed until no more
+        stars remain.  Note that in this case it is *possible* that the
+        loop will never end if the PSF has structure that causes
+        subtraction to create new sources infinitely.
+
+    Notes
+    -----
+    If there are problems with fitting large groups, change the
+    parameters of the grouping algorithm to reduce the number of sources
+    in each group or input a ``star_groups`` table that only includes
+    the groups that are relevant (e.g. manually remove all entries that
+    coincide with artifacts).
+
+    References
+    ----------
+    [1] Stetson, Astronomical Society of the Pacific, Publications,
+        (ISSN 0004-6280), vol. 99, March 1987, p. 191-222.
+        Available at: http://adsabs.harvard.edu/abs/1987PASP...99..191S
+    """
+
+    def __init__(self, group_maker, bkg_estimator, psf_model, fitshape,
+                 finder, fitter=LevMarLSQFitter(), niters=3,
+                 aperture_radius=None):
+        if finder is None:
+            raise ValueError("finder cannot be None. Please see the Detection, "
```

Comment:
Oh and isn't this unnecessary here because you've got it in the setter below?  So apply my comment here to L482, and remove this?